### PR TITLE
feat: i18n and localization documenso

### PR DESCRIPTION
**Pull request by anikdhabal**
- internationalized routing and locale detection
- offer dropdown to change language

currently supporting-

- English
- French
- Spanish
- German
- Italian
- Portuguese

### CODE REVIEW

**Comment by kodermax** [comment]
@anikdhabal Cool job, I would like to suggest a cookie-based language detection option.
Here are some advantages of changing the website language based on cookies rather than URLs like /en/:

Cleaner URLs - Removing the language tag from URLs results in simpler, cleaner, more user-friendly URLs that are easier to read and remember for visitors.

Better SEO - Eliminating duplicate URLs for different languages helps avoid splitting page equity between them, consolidating ranking signals for better SEO.

Easier navigation - Users can navigate the site in their language without having to constantly click language-tagged URLs. The experience feels more natural.

Greater flexibility - Cookies allow implementing more flexible logic, like auto-detecting user language, remembering language preferences across sessions, allowing switching languages quickly.

Avoiding thin content issues - Prevent duplicate low-value /en/ URLs that might be seen as "thin content" by search engines, leading to dilution of page rankings.

Internationalization - Cookies make it easier to expand the site to more languages without exponential URL growth. Language can be added just by adding a cookie value.

In summary, cookies provide a cleaner solution for handling multi-language websites, creating simpler URLs, improved SEO capabilities, better user experience, and greater flexibility for internationalization.

**Comment by anikdhabal** [comment]
> @anikdhabal Cool job, I would like to suggest a cookie-based language detection option. Here are some advantages of changing the website language based on cookies rather than URLs like /en/:
> 
> Cleaner URLs - Removing the language tag from URLs results in simpler, cleaner, more user-friendly URLs that are easier to read and remember for visitors.
> 
> Better SEO - Eliminating duplicate URLs for different languages helps avoid splitting page equity between them, consolidating ranking signals for better SEO.
> 
> Easier navigation - Users can navigate the site in their language without having to constantly click language-tagged URLs. The experience feels more natural.
> 
> Greater flexibility - Cookies allow implementing more flexible logic, like auto-detecting user language, remembering language preferences across sessions, allowing switching languages quickly.
> 
> Avoiding thin content issues - Prevent duplicate low-value /en/ URLs that might be seen as "thin content" by search engines, leading to dilution of page rankings.
> 
> Internationalization - Cookies make it easier to expand the site to more languages without exponential URL growth. Language can be added just by adding a cookie value.
> 
> In summary, cookies provide a cleaner solution for handling multi-language websites, creating simpler URLs, improved SEO capabilities, better user experience, and greater flexibility for internationalization.

@kodermax thanks for your suggestion. I was thinking about this before. Let see.

**Comment by dguyen** [comment]
@anikdhabal I'm not really familiar with the i18n libraries, did you generate the common.json translation files manually yourself?

File: apps/marketing/src/app/[locale]/(marketing)/open/page.tsx [hunk 1]
```diff
@@ -142,13 +145,12 @@ export default async function OpenPage() {
   return (
     <div className="mx-auto mt-6 max-w-screen-lg sm:mt-12">
       <div className="flex flex-col items-center justify-center">
-        <h1 className="text-3xl font-bold lg:text-5xl">Open Startup</h1>
+        <h1 className="text-3xl font-bold lg:text-5xl">{t('open_startup"')}</h1>
```
[the remark this answers is not in the document]
Comment (anikdhabal) [reply apps/marketing/src/app/[locale]/(marketing)/open/page.tsx:148]: yeah, thanks.

**Comment by anikdhabal** [comment]
> @anikdhabal I'm not really familiar with the i18n libraries, did you generate the common.json translation files manually yourself?

Yes, I generate those manually myself to ensure they are all correct.

**Comment by itechwebsolutions** [comment]
hi when changing the language in the /documents page.
the program is crashing, how can  we fix this bug?

**Comment by anikdhabal** [comment]
> hi when changing the language in the /documents page. the program is crashing, how can we fix this bug?

Hey, first of all, it's working fine for me. And these are not the final changes. I want to handle marketing and the main app differently. Currently, I am working on it and will finalize things soon.

**Comment by celevra** [comment]
if you need german translation i can help

**Comment by laurocesar** [comment]
Hi @anikdhabal 

How are you going here? I can help you with portuguese translation if you need it.

**Comment by TSP-Dev** [comment]
why documentso team has not considered this great work yet?, The more time passes, the more integration work is required.

**Comment by dguyen** [comment]
Hello everyone, thanks for all the contributions and offering to help for getting i18n into Documenso! 

Since i18n is something that bogs down development we've been wanting to do it in a way which reduces the load on the developer to maintain. With that in mind we decided to go with Lingui for the i18n integration, and Crowdin for the translations. 

Our plan is to roll it out to marketing first as a trial, then the changes for the web application will be rolled out after.

More details and reasoning about our choices can be found here

https://github.com/documenso/documenso/pull/1223

We appreciate the community effort and help for the i18n push, and we apologize for taking a while to get it ready.
